### PR TITLE
sqlite: use bundled version to enable FK checks

### PR DESCRIPTION
**Pull request by hackaugusto**
I'm not sure why the PRAGMA wasn't sufficient. But this was caught by @phklive when integrating the faucet. The feature flag seems to be required

### CODE REVIEW

File: store/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,7 @@ miden-node-utils = { path = "../utils", version = "0.2" }
 miden-objects = { workspace = true }
 once_cell = { version = "1.18.0" }
 prost = { version = "0.12" }
-rusqlite = { version = "0.30", features = ["array", "buildtime_bindgen"] }
+rusqlite = { version = "0.30", features = ["array", "buildtime_bindgen", "bundled"] }
```
Comment (bobbinth) [inline store/Cargo.toml:33]: Should we bump-up the version to v0.31 (I think that's the latest one now).

Comment (hackaugusto) [reply store/Cargo.toml:33]: Not for the moment, Deadpool has the SQLite version pinned

Comment (phklive) [inline store/Cargo.toml:33]: This is the feature used by the Miden client. I am curious if the bug could be related, needing to have the same features enabled on all crates.

File: store/src/db/migrations.rs [hunk 2]
```diff
@@ -50,16 +50,16 @@ pub static MIGRATIONS: Lazy<Migrations> = Lazy::new(|| {
...
+            CONSTRAINT nullifiers_block_number_is_u32 CHECK (block_number >= 0 AND block_number < 4294967296)
         ) STRICT, WITHOUT ROWID;
         ",
     )])
 });
 
 #[test]
 fn migrations_test() {
-    assert!(MIGRATIONS.validate().is_ok());
+    assert_eq!(MIGRATIONS.validate(), Ok(()));
```
Comment (phklive) [inline store/src/db/migrations.rs:64]: Why is this change needed? Isn't it equivalent?

File: store/src/db/tests.rs [hunk 3]
```diff
@@ -18,12 +18,36 @@ fn create_db() -> Connection {
...
+    let block_header = BlockHeader::new(
+        num_to_rpo_digest(1),
+        block_num,
+        num_to_rpo_digest(3),
+        num_to_rpo_digest(4),
+        num_to_rpo_digest(5),
+        num_to_rpo_digest(6),
+        num_to_rpo_digest(7),
+        num_to_rpo_digest(8),
+        9_u8.into(),
```
Comment (phklive) [inline store/src/db/tests.rs:34]: Why are we using 9_u8.into() here instead of num_to_rpo_digest ?

File: store/src/db/tests.rs [hunk 4]
```diff
@@ -382,6 +416,9 @@ fn test_db_account() {
 fn test_notes() {
     let mut conn = create_db();
 
+    let block_num1 = 1;
```
Comment (phklive) [inline store/src/db/tests.rs:419]: Would the idiomatic way to write this var name be `block_num_1` ?

File: store/src/db/tests.rs [hunk 5]
```diff
@@ -434,14 +470,17 @@ fn test_notes() {
         &mut conn,
         &[(tag >> 48) as u32],
         &[],
-        block_num - 1,
+        block_num1 - 1,
     )
     .unwrap();
     assert_eq!(res, vec![note.clone()]);
 
+    let block_num2 = note.block_num + 1;
```
Comment (phklive) [inline store/src/db/tests.rs:478]: Same here regarding idiomatic naming.

**Review by phklive — APPROVED** [review]
Added a few questions and nits, overall LGTM.

File: store/src/db/migrations.rs (hunk 2 above)
Comment (hackaugusto) [reply store/src/db/migrations.rs:64]: the result of `validate()` is a bool, that erases the error message. Comparing against `Ok(())` preserver the sqlite3 error if the test fails

File: store/Cargo.toml (hunk 1 above)
Comment (hackaugusto) [reply store/Cargo.toml:33]: which bug do you mean? This by itself wasn't a bug in itself, but it allowed one to slip through the cracks (the name mismatch on the other PR).

File: store/src/db/tests.rs (hunk 3 above)
Comment (hackaugusto) [reply store/src/db/tests.rs:34]: I _think_ we don't have a `Into` conversion from u8 to RpoDigest. I think this is just wrapping the type to create the block header

Comment (phklive) [reply store/src/db/tests.rs:34]: What do you call `wrapping` ?

File: store/src/db/tests.rs (hunk 4 above)
Comment (phklive) [reply store/src/db/tests.rs:419]: I prefer it the idiomatic way personally.

File: store/src/db/tests.rs (hunk 3 above)
Comment (hackaugusto) [reply store/src/db/tests.rs:34]: when you have a unit type struct, wrapping an inner type.

File: store/src/db/tests.rs (hunk 4 above)
Comment (hackaugusto) [reply store/src/db/tests.rs]: changed
